src/scripts/index.ts

```diff
@@ -10,16 +11,18 @@ import { newAdapterCommand } from './newAdapterCommand'
 
 const program = new Command('mmi-adapters')
 
-featureCommands(program)
+const defiAdapers = new DefiAdapter()
+
+featureCommands(program, defiAdapers)
 
 newAdapterCommand(program)
 
-blockAverage(program)
+blockAverage(program, defiAdapers)
 
 buildContractTypes(program)
 
-buildMetadata(program)
+buildMetadata(program, defiAdapers)
 
-buildSnapshots(program)
+buildSnapshots(program, defiAdapers)
 
```

Comment:
Here Im passing around one instance, let me know if you have a preferred solution. Injecting class into function might have some benefit in future.